zilsd.adoc

```diff
@@ -78,4 +118,182 @@ An implementation may have a requirement to issue a load/store pair instruction
 
 If the core implementation does not support Zilsd instructions to non-idempotent memories, the core may use an idempotency PMA to detect it and take a load or store access fault exception in order to avoid unpredictable results.
 
-Software should only use these instructions on non-idempotent memory regions when software can tolerate the required memory accesses being issued repeatedly in the case that they cause exceptions.
\ No newline at end of file
+Software should only use these instructions on non-idempotent memory regions when software can tolerate the required memory accesses being issued repeatedly in the case that they cause exceptions.
+
+<<<
+
+=== Instructions
+[#insns-ld,reftext="Load doubleword to register pair, 32-bit encoding"]
+==== ld
+
+Synopsis::
+Load doubleword to even/odd register pair, 32-bit encoding
+
+Mnemonic::
+ld rd, simm(rs1)
+
+Encoding (RV32)::
+[wavedrom, ,svg]
+....
+{reg: [
+  {bits: 7,  name: 'opcode',    attr: ['7', 'LOAD'], type: 8},
+  {bits: 5,  name: 'rd', attr: ['5', 'dest, dest[0]=0'], type: 2},
+  {bits: 3,  name: 'funct3', attr: ['3', 'width=D'], type: 8},
+  {bits: 5,  name: 'rs1', attr: ['5', 'base'], type: 4},
+  {bits: 12, name: 'imm[11:0]', attr: ['12', 'offset[11:0]'], type: 3},
+]}
+....
+
+Description:: 
+Loads a 64-bit value into registers `rd` and `rd+1`.
+The effective address is obtained by adding register rs1 to the
+sign-extended 12-bit offset.
+
+Included in: <<zilsd>>
+
+<<<
+
+[#insns-sd,reftext="Store doubleword from register pair, 32-bit encoding"]
+==== sd
+
+Synopsis::
+Store doubleword from even/odd register pair, 32-bit encoding
+
+Mnemonic::
+sd rs2, simm(rs1)
+
+Encoding (RV32)::
+[wavedrom, ,svg]
+....
+{reg: [
+  {bits: 7,  name: 'opcode',   attr: ['7', 'STORE'], type: 8},
+  {bits: 5,  name: 'imm[4:0]', attr: ['5', 'offset[4:0]'], type: 3},
+  {bits: 3,  name: 'funct3', attr: ['3', 'width=D'], type: 8},
+  {bits: 5,  name: 'rs1', attr: ['5', 'base'], type: 4},
+  {bits: 5,  name: 'rs2', attr: ['5', 'src, src[0]=0'], type: 4},
+  {bits: 7, name: 'imm[11:5]', attr: ['7', 'offset[11:5]'], type: 3},
+]}
+....
+
+Description:: 
+Stores a 64-bit value from registers `rs2` and `rs2+1`.
+The effective address is obtained by adding register rs1 to the
+sign-extended 12-bit offset.
+
+Included in: <<zilsd>>
+
+<<<
+
+[#insns-cldsp,reftext="Stack-pointer based load doubleword to register pair, 16-bit encoding"]
+==== c.ldsp
+
+Synopsis::
+Stack-pointer based load doubleword to even/odd register pair, 16-bit encoding
+
+Mnemonic::
+c.ldsp rd, uimm(sp)
+
+Encoding (RV32)::
+[wavedrom, ,svg]
+....
+{reg: [
+  {bits: 2, name: 'op',     type: 8, attr: ['2','C2']},
+  {bits: 5, name: 'imm',    type: 3, attr: ['5','offset[4:3|8:6]']},
+  {bits: 5, name: 'rd',     type: 2, attr: ['5','dest≠0, dest[0]=0']},
+  {bits: 1, name: 'imm',    type: 3, attr: ['1','offset[5]']},
+  {bits: 3, name: 'funct3', type: 8, attr: ['3','C.LDSP']},
```

Comment:
the value of this funct3 isn't specified anywhere, and similarly for the other instructions.
and the ≠ character doesn't render in the pdf.
I really think the encoding page needs to specify every bit in the binary encoding, including things like width=D, otherwise you have to look in another doc to sew what the encoding is (or riscv-opcodes)

Reply:
i have updated the encoding to explicit in a025144.
I have no idea why the ≠ character does not render. 
It might be my local setup, because it is done in the same way for C and there it renders.

Reply:
The not-equal sign was not rendering because the sans-serif font inclusion was commented out in the theme.  I've updated docs-resources to fix this.  I'll open an issue to update the docs-resources pointer.

Reply:
thanks